x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/DownsampleAction.java

```diff
@@ -230,24 +253,30 @@ public List<Step> toSteps(Client client, String phase, StepKey nextStepKey) {
         ClusterStateWaitUntilThresholdStep downsampleAllocatedStep = new ClusterStateWaitUntilThresholdStep(
             new WaitForIndexColorStep(
                 waitForDownsampleIndexKey,
-                copyMetadataKey,
+                forceMergeIndex ? forceMergeKey : copyMetadataKey,
                 ClusterHealthStatus.YELLOW,
-                (indexName, lifecycleState) -> lifecycleState.downsampleIndexName()
+                DOWNSAMPLED_INDEX_NAME_SUPPLIER
             ),
             cleanupDownsampleIndexKey
         );
+        ForceMergeStep forceMergeStep = forceMergeIndex
+            ? new ForceMergeStep(forceMergeKey, waitForSegmentCountKey, client, 1, DOWNSAMPLED_INDEX_NAME_SUPPLIER)
+            : null;
+        SegmentCountStep segmentCountStep = forceMergeIndex
+            ? new SegmentCountStep(waitForSegmentCountKey, copyMetadataKey, client, 1, DOWNSAMPLED_INDEX_NAME_SUPPLIER)
+            : null;
```

Comment:
I don't think we need to nullify these variables if `forceMergeIndex` is false. We already have the `if (forceMergeIndex) {` check below, so this feels a bit redundant.